book/src/Setup.md

```diff
@@ -1,64 +1,125 @@
 # Setup
+This document has been updated for the latest version of Kubernetes v1.11.
+
+
+## Enable privileged Pods
 
 To use CSI drivers, your Kubernetes cluster must allow privileged pods (i.e. `--allow-privileged` flag must be set to `true` for both the API server and the kubelet). This is the default in some environments (e.g. GCE, GKE, `kubeadm`).
 
-Moreover, as stated in the [mount propagation docs][mount-propagation-docs], the Docker daemon of the cluster nodes must allow shared mounts. [This page][docker-shared-mount] explains how to check if shared mounts are enabled and how to configure Docker for shared mounts.
+Ensure your API server are started with the privileged flag:
 
-This document has been updated to the latest version of Kubernetes v1.11.
+```shell
+$ ./kube-apiserver ...  --allow-privileged=true ...
+```
+
+```shell
+$ ./kubelet ...  --allow-privileged=true ...
+```
 
-## Alpha Features
+## Enabling mount propagation
+Another feature that CSI depends on is mount propagation.  It allows the sharing of volumes mounted by one container with other containers in the same pod, or even to other pods on the same node.  For mount propagation to work, the Docker daemon for the cluster must allow shared mounts. See the [mount propagation docs][mount-propagation-docs] to find out how to enable this feature for your cluster.  [This page][docker-shared-mount] explains how to check if shared mounts are enabled and how to configure Docker for shared mounts.
 
-#### CSI Raw block volume support
+## Enable raw block volume support (alpha)
 
-[CSI Raw block volume support][rawvol]: To enable support for raw block volumes
-you must set the following feature gate on Kubernetes v1.11:
+Kubernetes now has [raw block volume Support][rawsupport] as an alpha implementation. If you want to use the 
+[CSI raw block volume support][rawvol], you must enable the feature (for your Kubernetes binaries including server, kubelet, controller manager, etc) with the  `feature-gates` flag as follow:
 
 ```
---feature-gates=BlockVolume=true,CSIBlockVolume=true
+$ kube<binary> --feature-gates=BlockVolume=true,CSIBlockVolume=true ...
 ```
 
-Also, see [Raw Block Volume Support][rawsupport]
+## CSI driver custom resources (alpha)
+Starting with version 1.12, two new custom resource definitions (or CRDs) have been introduced as a way to augment the configuration of CSI drivers.  The `CSRDriver` resource is a representation of a driver registered with the cluster. The purpose of the CSIDriver is to allow administrator to specify and update CSI configurations using Kubernetes resource constructs.  
 
-#### Kubelet Plugin Watcher
+```yaml
```

Comment:
I'm not sure you need this here anymore since the driver-registrar will add/create it. I think if we way "The driver-registrar can register the driver for you or you can do it by hand using the following spec.." Something like that.

Reply:
@lpabon I will update the doc with a sample CSIDriver object just to show what it looks like.